src/gateway/ws.rs

```diff
@@ -101,13 +108,72 @@ const TIMEOUT: Duration = Duration::from_millis(500);
 const DECOMPRESSION_MULTIPLIER: usize = 3;
 
 impl WsClient {
-    pub(crate) async fn connect(url: Url) -> Result<Self> {
+    async fn connect_with_proxy_async(
```

Comment:
This whole method should be replaced with something using `reqwest`, which is already in the library.

Reply:
> This whole method should be replaced with something using `reqwest`, which is already in the library.

Reqwest doesn't expose API that can be used in websocket proxy. But there is an implementation that borrows from reqwest's design.
https://github.com/snapview/tungstenite-rs/issues/177#issuecomment-1352808621

Reply:
Just like [Itsusinn](https://github.com/Itsusinn) said. I replaced the http connect code with async-http-proxy[https://crates.io/crates/async-http-proxy] which is mentioned in that issue.